### PR TITLE
motherlode: Perform MLM location check when the player is logging in or entering an area

### PR DESCRIPTION
**Pull request by Quasindro**

### CODE REVIEW

**Comment by Quasindro** [comment]
fixes #9301

File: runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java [hunk 1]
```diff
@@ -376,6 +376,8 @@ public void onGameStateChanged(GameStateChanged event)
 			// on region changes the tiles get set to null
 			veins.clear();
 			rocks.clear();
+
+			inMlm = checkInMlm();
```
Comment (Adam-) [inline runelite-client/src/main/java/net/runelite/client/plugins/motherlode/MotherlodePlugin.java]: If this is being moved to here, you don't need it below in the `LOGGED_IN` case

**Comment by Adam-** [comment]
Do you know why this fixes the issue? Is it that the spawn events happen during loading? I wonder how this ever worked..

**Comment by Quasindro** [comment]
Yeah, the obstacles seem to have been an afterthought. When I checked all the *spawned events and compared them with the gamestates, it seems that the wallobjects spawn correctly after the GameState is set to LOGGED_IN and the checkInMlm() method has already ran. The gameobjects, however, are being loaded when the client is still loading.

Committed the change suggested in a review. You were right, removing that check did not change anything.

**Pull request merged**